src/server/protocol_client.h

```diff
@@ -83,6 +83,7 @@ class ProtocolClient {
   // is done with the result of the call; Calling ConsumeInput may invalidate the data in the result
   // if the buffer relocates.
   io::Result<ReadRespRes> ReadRespReply(base::IoBuf* buffer = nullptr, bool copy_msg = true);
+  io::Result<ReadRespRes> ReadRespReply(uint32_t timeout);
```

Comment:
not the most elegant way to add a timeout... 🙂

Reply:
What do you suggest?

Reply:
I can improve in the next PR